### PR TITLE
Add ipvs e2e test job

### PR DESCRIPTION
**Pull request by Lion-Wei**
We were working on move IPVS outside of alpha for the 1.9 release. see this issue: https://github.com/kubernetes/kubernetes/issues/51602.
This pr is about to create a new e2e test job for IPVS, which mentioned in this issue: https://github.com/kubernetes/kubernetes/issues/52834.

### CODE REVIEW

**Comment by krzyzacy** [comment]
I think this should be a ci job.
/assign @MrHohn

File: jobs/config.json [hunk 1]
```diff
@@ -9412,6 +9412,28 @@
...
+      "--ginkgo-parallel=30",
+      "--mode=docker",
+      "--provider=gce",
+      "--stage=gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gce",
+      "--test_args=--ginkgo.focus=\\[sig-network\\] --ginkgo.skip=\\[Slow\\]|\\[Serial\\]|\\[Disruptive\\]|\\[Flaky\\]|\\[Feature:.+\\] --minStartupPods=8",
+      "--timeout=60m"
+    ],
+    "scenario": "kubernetes_e2e",
+    "sigOwners": [
+      "UNKNOWN"
```
Comment (fejta) [inline jobs/config.json]: Please find a sig willing to sponsor the maintenance of this job

Comment (Lion-Wei) [reply jobs/config.json]: sorry, my mistake

Comment (Lion-Wei) [reply jobs/config.json]: fixed. Thanks for reminding!

File: prow/config.yaml [hunk 2]
```diff
@@ -955,6 +955,68 @@ presubmits:
     branches:
     - release-1.6
 
+  - name: pull-kubernetes-e2e-gce-ipvs
+    agent: kubernetes
+    skip_branches:
+    - release-1.6
```
Comment (BenTheElder) [inline prow/config.yaml]: please add a comment next to these explaining why for anyone reading this (presumably:  `# IPVS wasn't available before 1.8`)

File: jobs/config.json [hunk 3]
```diff
@@ -9412,6 +9412,28 @@
...
+      "--build",
+      "--cluster=",
+      "--env-file=jobs/platform/gce.env",
+      "--env-file=jobs/env/pull-kubernetes-e2e.env",
+      "--env-file=jobs/env/pull-kubernetes-e2e-gce-ipvs.env",
+      "--extract=local",
+      "--gcp-node-image=gci",
+      "--gcp-zone=us-central1-f",
+      "--ginkgo-parallel=30",
+      "--mode=docker",
```
Comment (BenTheElder) [inline jobs/config.json]: `--mode=docker` does not work on Prow since kubernetes doesn't really support running docker commands from your pods, please use `--build=bazel` and drop the `--mode` flag

**Comment by BenTheElder** [comment]
you'll also need to add matching `kubernetes-security` presubmit entries in the prow config.yaml if this stays as a kubernetes presubmit

**Comment by krzyzacy** [comment]
/hold
please state why this need to be a presubmit job

File: prow/config.yaml (hunk 2 above)
Comment (Lion-Wei) [reply prow/config.yaml]: Sure, thanks.

**Comment by Lion-Wei** [comment]
@krzyzacy  Sure, we always need check if IPVS proxy mode works well for each ipvs or kube-proxy related pr. Since ipvs is a growing feature, there might be lot pr about IPVS, so I prefer make this a presubmit job. Besides, there are only 35 tests, wouldn't take lots time. : )

**Comment by xiangpengzhao** [comment]
IMO, making IPVS related test cases run in the existing job `pull-kubernetes-e2e-gce` would be enough. It's not so necessarily to be a separate job.

**Comment by MrHohn** [comment]
> IMO, making IPVS related test cases run in the existing job pull-kubernetes-e2e-gce would be enough. It's not so necessarily to be a separate job. [maybe I miss sth here]

I think IPVS test cases shouldn't be included in `pull-kubernetes-e2e-gce`:
1. It might be risky to make an alpha feature blocking submit queue.
2. We still need to guarantee kube-proxy iptables mode is working properly.

> I think this should be a ci job.

+1, presubmit job is more about running sanity-checking tests and usually these tests would be a subset of what we test in CIs. Creating a feature specific presubmit job seems to have too much overhead. Ref current [presubmit dashboard](https://k8s-testgrid.appspot.com/presubmits-kubernetes).

**Comment by krzyzacy** [comment]
@Lion-Wei it cost an entire cluster to run a presubmit which is not trivial. If you agree with @MrHohn please convert it to a ci/periodic job, you can reference `ci-kubernetes-e2e-gci-gce-ip-alias` for example :-)

File: jobs/env/ci-kubernetes-e2e-gci-gce-ipvs.env [hunk 4]
```diff
@@ -0,0 +1,10 @@
+### job-env
+# Use proxy mode IPVS
+KUBE_FEATURE_GATES=SupportIPVSProxyMode=true
+KUBEPROXY_TEST_ARGS=--proxy-mode=ipvs
+
+# Enable the PodSecurityPolicy in tests.
+ENABLE_POD_SECURITY_POLICY=true
+
+CREATE_CUSTOM_NETWORK=true
```
Comment (MrHohn) [inline jobs/env/ci-kubernetes-e2e-gci-gce-ipvs.env]: We don't need to create custom network in CIs.

File: jobs/config.json [hunk 5]
```diff
@@ -3808,6 +3808,25 @@
       "sig-network"
     ]
   },
+  "ci-kubernetes-e2e-gci-gce-ipvs": {
+    "args": [
+      "--check-leaked-resources",
+      "--env-file=jobs/platform/gce.env",
+      "--env-file=jobs/env/pull-kubernetes-e2e.env",
```
Comment (MrHohn) [inline jobs/config.json]: We shouldn't need `pull-kubernetes-e2e.env`?

File: jobs/env/ci-kubernetes-e2e-gci-gce-ipvs.env [hunk 6]
```diff
@@ -0,0 +1,10 @@
+### job-env
+# Use proxy mode IPVS
+KUBE_FEATURE_GATES=SupportIPVSProxyMode=true
+KUBEPROXY_TEST_ARGS=--proxy-mode=ipvs
+
+# Enable the PodSecurityPolicy in tests.
+ENABLE_POD_SECURITY_POLICY=true
```
Comment (MrHohn) [inline jobs/env/ci-kubernetes-e2e-gci-gce-ipvs.env]: Why is this needed?

File: jobs/env/ci-kubernetes-e2e-gci-gce-ipvs.env (hunk 4 above)
Comment (Lion-Wei) [reply jobs/env/ci-kubernetes-e2e-gci-gce-ipvs.env]: actually, I was referenced `pull-kubernetes-e2e-gce` :( , thanks for reminding.

**Comment by MrHohn** [comment]
Thanks, this LGTM now.

/lgtm
